### PR TITLE
Until

### PR DESCRIPTION
**Pull request by mcdeoliveira**
This is an implementation of the manyTill method mentioned in Issue #14.

It adds functionality similar to the PR #31, which seems to have stalled, but this time implemented as a method. 

It has options for min, max and consume_other, which is False by default.

### CODE REVIEW

**Review by spookylukey — COMMENTED** [review]
Thanks so much for the PR, this looks like a great addition, and the careful unit tests for edge cases etc. are much appreciated.

I have some questions, see what you think.

Also, I've just recently added black to our set of tools, it would help if you could install pre-commit to apply all the linters/formatters - see https://parsy.readthedocs.io/en/latest/contributing.html  Otherwise I can do it.

File: src/parsy/__init__.py [hunk 1]
```diff
@@ -181,6 +181,53 @@ def at_least(self, n):
     def optional(self):
         return self.times(0, 1).map(lambda v: v[0] if v else None)
 
+    def until(self, other, min=1, max=float('inf'), consume_other=False):
+
+        @Parser
+        def until_parser(stream, index):
+            # behaves as times if parser is not defined
+            if not other:
+                return self.times(min, max)(stream, index)
```
[the remark this answers is not in the document]
Comment (mcdeoliveira) [reply src/parsy/__init__.py:190]: Not having it checked is fine.

File: src/parsy/__init__.py [hunk 2]
```diff
@@ -181,6 +181,53 @@ def at_least(self, n):
...
+                    return Result.success(index, values)
+
+                # exceeded max?
+                if times >= max:
+                    # return failure, it matched parser more than max times
+                    return Result.failure(index,
+                                          f'at most {max} items')
+
+                # failed, try parser
+                result = self.__call__(stream, index).aggregate(result)
```
[the remark this answers is not in the document]
Comment (mcdeoliveira) [reply src/parsy/__init__.py:214]: Yes, i was just being (too) verbose.

**Comment by spookylukey** [comment]
Thanks so much, I merged a squashed version of this with black applied and a few small tweaks.
